sdk/python/kfp/components/types/type_utils.py

```diff
@@ -349,34 +352,12 @@ def __exit__(self, *unused_args) -> None:
 
 # for reading in IR back to in-memory data structures
 IR_TYPE_TO_IN_MEMORY_SPEC_TYPE = {
-    'STRING':
-        'String',
-    'NUMBER_INTEGER':
-        'Integer',
-    'NUMBER_DOUBLE':
-        'Float',
-    'LIST':
-        'List',
-    'STRUCT':
-        'Dict',
-    'BOOLEAN':
-        'Boolean',
-    artifact_types.Artifact.schema_title:
```

Comment:
This previously existed to help with handling the two representations of artifacts we were passing around: `system.Artifact` and `Artifact`. We no longer need this.